lib/animina/accounts/resources/photo.ex

```diff
@@ -175,4 +195,75 @@ defmodule Animina.Accounts.Photo do
       reference :user, on_delete: :delete
     end
   end
+
+  def create_optimized_photos(record) do
+    case check_if_image_magick_is_installed() do
+      {:ok, _} ->
+        resize_images_with_imagemagick(record)
+
+      {:error, _} ->
+        copy_images_directly(record)
+    end
+  end
+
+  defp copy_images_directly(record) do
+    for type <- [:thumbnail, :normal, :big] do
+      OptimizedPhoto.create(%{
+        image_url: record.filename,
+        type: type,
+        user_id: record.user_id,
+        photo_id: record.id
+      })
+    end
+  end
+
+  # TODO: Confirm if the dimensions are  okay for each type
+  defp resize_images_with_imagemagick(record) do
```

Comment:
Please add that the result will be encoded as a `webp`. We can save a bit more bandwidth with that.

Reply:
Done